Documentation/tectonic.md

```diff
@@ -0,0 +1,47 @@
+# Using Rook on Tectonic Bare Metal
+
+Here is a running guide on how to implement Rook on Tectonic. A complete guide on how to install Tectonic is out of the scope of the Rook project. But many info can be found on the [Tectonic website](https://coreos.com/tectonic/docs/latest/)
+
+## Prequisites
+
+- An installed tectonic-installer. These steps are described on [the Tectonic website](https://coreos.com/tectonic/docs/latest/install/bare-metal/#4-tectonic-installer)
+- A running matchbox node which will do the provisioning (Matchbox is only required if you are running Tectonic on Bare metal)
+- You can run through all steps of the GUI installer, but in the last step, choose `Boot manually`. This way we can make the necessary changes first.
+
+## Edit the kubelet.service file
+
+We need to make a few adaptions to the Kubelet systemd service file generated by the Tectonic-installer.
+
+First change to the directory in which you untarred the tectonic installer and find your newly generated cluster configuration files.
+
+```bash
+cd ~/tectonic/tectonic-installer/LINUX-OR-DARWIN/clusters
+```
+
+
+Open the file `modules/ignition/resources/services/kubelet.service` in your favorite editor and after the last line containing `ExecStartPre=...`, paste the following extra lines:
+
+```bash
+ExecStartPre=/bin/mkdir -p /var/lib/kubelet/volumeplugins
```

Comment:
are these steps necessary? The agent already creates that directory in https://github.com/rook/rook/blob/master/pkg/agent/flexvolume/server.go#L131.
Also i think rook also creates `/var/lib/rook` since it hostmount that. Can you verify since im dont have a tectonic environment?

Reply:
This is done because Tectonic doesn't automatically boots Rook. But during the setup we already adapt the Kubelet so the user doesn't need to change anything to the `kubelet.service` file on each node afterwards to be able to run Rook.
So the directory `/var/lib/kubelet/volumeplugins` must be created for the Kubelet to start, otherwise it will throw an error that  `--volume-plugin-dir=/var/lib/kubelet/volumeplugins` is invalid.

Reply:
@kokhang I believe the relevant issue is https://github.com/poseidon/typhoon/issues/57, because of which kubelet has to be customized at the installation of coreos

Reply:
Oh you have actually posted the original bug report - https://github.com/coreos/bugs/issues/2201

Reply:
@adarshaj Are you sure you can run the cluster without the Kubelet customizations? If I remember well it didn't work on my dev-server about a month ago. If it would work, this whole PR seems completely unnecessary (a)

Reply:
@adarshaj right. the default plugin of `/usr/libexec/...` is read only in coreos. But i was refererncing the explicit creation of the `/var/lib/kubelet/volumeplugins` dir. Since `/var/lib/kubelet` is not readonly, the agent should succeed at creating it.

@hacor  I remember that i used matchbox a few months back to deploy kubernetes and the only thing i did was to set the `-volume-plugin-dir=/var/lib/kubelet/volumeplugins` but that could've changed. I just wanted to ensure if this was still the case. Thanks for looking into it.

Reply:
@hacor Is the `mkdir` still needed?

Reply:
AFAIK Kubernetes 1.7.9 doesn't create this directory itself yet. I can't find any other proof it does in newer versions. This is needed here for several reasons:
- This directory needs to exist for the kubelet to be able to start with the current `--volume-plugin` option
- The Kubelet will fail if it doesn't exist but the `--volume-plugin` option is provided
- Therefor when the directory is created by the Rook agent, it seems logical that the Kubelet itself is already up and running. So for Tectonic, this feature is not an advantage (I guess it is for other setups)